mozilla-platform-ops/ronin_puppet
# Ubuntu 18.04, round 2

based on https://github.com/mozilla-platform-ops/ronin_puppet/pull/217

DONE:
- sudoers fixing and testing
- install/manage puppet 6
- puppet run at boot (modules/puppet/manifests/atboot.pp)
- modules/gui (configure xvfb for autologin)
- disable services
  - gdm / Xwayland via systemd default
- Xsession
  - g-w start
- locale
- updated to latest g-w/tc components
- fix/add talos stuff
  - BC says http proxy not required, tests seem to confirm
- disable services
  - gnome-initial-setup
  - deja-dup-monitor, update-notifier
  - those mentioned in modules/disableservices in build-puppet
- added imagemagick 7
- added psutil (https://bugzilla.mozilla.org/show_bug.cgi?id=1667166)
- chrome autoupdate
- write more spec tests
- updated to latest g-w/tc components, again
- added tc start-worker
- use tc start-worker / worker-runner
- operator hold mode (similar to quarantine)
- securitize module
  - remove root pw
  - remove relops pw
  - add relops users ssh keys to relops user's authorized keys
  - relops can sudo all without pw
- mercurial upgrade/debugging
- grub timeout to 10 seconds
- compare services, processes to 1604 on moonshot
- compare packages to 1804 docker image
- change worker type in gecko_t_linux_talos_generic_worker.pp

TESTING:
- testing performed on t-linux64-ms-573
  - puppet atboot convergence on metal
  - generic-worker starts
  - vnc works (connected successfully via ssh tunnel)
- tc testing
    - round one: https://treeherder.mozilla.org/#/jobs?repo=try&group_state=expanded&revision=b932818471157737c2c2ab4351d2d7f63e057895
      - results look good
    - round two: https://treeherder.mozilla.org/#/jobs?repo=try&group_state=expanded&resultStatus=superseded%2Ctestfailed%2Cbusted%2Cexception%2Csuccess%2Cretry%2Cusercancel%2Crunning%2Cpending%2Crunnable&tier=1%2C2%2C3&revision=e9df3ce8632c777f20058835e92e9c1b99af23ed

TODO:
- nothing before landing

TODO (after merging this PR):
-  modules/roles_profiles/manifests/profiles/gecko_t_linux_talos_generic_worker.pp: point at master, vs this branch
    - can't do in this PR as the role already exists on master and it would revert the currently running test hosts

FUTURE PRS:
- syslog/papertrail
- auditd
- telegraf
- repo pinning?
  - what repos are enabled?
- nrpe checks?
- firewall fixing
  - copy from puppetagain, low priority
